File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/logic/BotLogicQuick.java
package cz.muni.fi.umimecesky.logic;

import cz.muni.fi.umimecesky.pojo.RaceConcept;
import cz.muni.fi.umimecesky.utils.Util;

public class BotLogicQuick implements RobotLogic {

    private RaceConcept concept;

    public BotLogicQuick(RaceConcept concept) {
        this.concept = concept;
    }

    @Override
    public int millisecondsPerSolution() {
        return 3000 - (int) (1800 * concept.levelProgress());
    }

    @Override
    public double correctnessRatio() {
        return 0.65 + Util.roundBy2places(0.3 * concept.levelProgress());
    }

    @Override
    public int hopsPerCorrect() {
        return 1;
    }

}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/logic/BotLogicQuick.kt
package cz.muni.fi.umimecesky.logic

import cz.muni.fi.umimecesky.logic.RobotLogic.Companion.roundBy2places
import cz.muni.fi.umimecesky.pojo.RaceConcept

class BotLogicQuick(private val concept: RaceConcept) : RobotLogic {

    override fun millisecondsPerSolution(): Long = 3000L - (1800L * concept.levelProgress()).toLong()

    override fun correctnessRatio(): Double =
            0.65 + roundBy2places(0.3 * concept.levelProgress())

    override fun hopsPerCorrect(): Int = 1

}
